common/download.go

```diff
@@ -203,7 +204,9 @@ func (d *DownloadClient) Get() (string, error) {
 
 // VerifyChecksum tests that the path matches the checksum for the
 // download.
-func (d *DownloadClient) VerifyChecksum(path string) (bool, error) {
+func (d *DownloadClient) VerifyChecksum(path string, state multistep.StateBag) (bool, error) {
```

Comment:
My only nitpick is that I think it'll be better to pass the ui directly instead of passing the statebag here.  Otherwise, I love this change!